### ISSUE TITLE
Compute response times metrics

### ISSUE THREAD

**Issue by futurechris** [hackoregon/emergency-response#26]
Variety of metrics Mark gave us that they use:

- [ ] Effective Team ResponseTime: Latest Onscene - Earliest Dispatched
- [ ] TurnoutTime: Enroute-Dispatched
- [x] TravelTime: Onscene-Enroute
- [x] ResponseTime: Onscene-Dispatched
- [ ] CommittedTime: Clear-Dispatched
- [ ] TotalOnSceneTime: Clear-Onscene
- [ ] ReturnToQuartersTime: InQuarters-Clear
- [ ] TimeWithPatient: Clear-WithPatient
- [ ] TimeToPatient: WithPatient-Onscene
- [ ] TotalTimeToPatient: withPatient-Dispatched
- [ ] CallProcessingTime: Dispatched-Received (or Initiated or Entered, whatever was earliest)

For now it's probably sufficient that we just work on these as a checklist, but we could split it out into individual tickets if they get big enough.

**Comment by kielejocain** [comment]
Blocked by #1

**Comment by sky-t** [comment]
During the 1/25 meeting Mark also mentioned an interest in response time vs type of incident. Specifically, it would be interesting for PF&R to understand whether response time differs if the incident is a medical call vs a fire call. Theoretically, PF&R should be at least be slightly faster for medical calls since the equipment requirements are fewer but there is a perception that fire responders may have a bias toward fire incidents.

**Comment by futurechris** [comment]
Just adding myself in. I've been working on "TotalOnSceneTime" when I've found time over the last couple days, though what I'm working on should be able to produce most of these. I understand @matthino is doing something similar, so hopefully we can compare notes at tonight's meeting.

**Comment by hannahkane** [comment]
Please also compute these per FMA

**Comment by futurechris** [comment]
@markawhitaker I ran into a question when computing TotalOnSceneTime: Should we calculate it for each responder or per incident? Both?

E.g. incident_id 855800 has two responders.
OnScene times are respectively 19:13:53 and 19:21:10
Clear times are respectively 19:30:49 and 20:14:05.
If TotalOnSceneTime is from (first on scene) to (last clear), then it would be 1 hour 8 seconds.
If it's per-responder, then we have two times. One is 16 min 56 seconds, the other is 53 min 5 seconds.

Which does PF&R care about?

A follow-up question: How do later investigations get handled?

If we do calculation from first onscene to last clear, and an investigation happened weeks later, it creates an extremely long TotalOnScene value.
Even if not doing the combined TotalOnScene, should investigator times be included in this calculation at all? It doesn't seem like investigator time on scene should be treated the same as time spent during an actual incident.

Thanks!

**Comment by markawhitaker** [comment]
@futurechris Not to be difficult, but both could be useful depending on the analysis. If we're looking at total workload or availability for individual companies, then total on-scene time for each separate apparatus would be useful, and every unit response would have a separate total on-scene time.  But in another analysis, (such as a medical call where the goal is to usually assess, stabilize, and transport as quickly as possible) we may be interested in the collective on-scene time, where each incident has just one total on-scene time.

I would drop investigator units from all time metrics. They're not considered part of PF&R's "emergency response" and their response time, on-scene time is typically not critical.

**Comment by futurechris** [comment]
Makes total sense, thanks.

I think we'll start with collective time, since it's a more straightforward metric to compute. Then we'll add individual metrics as we're able, and as our analytical tools mature.
